marshmallow_recipe/metadata.py

```diff
@@ -27,6 +28,19 @@ def __repr__(self) -> str:
     def __str__(self) -> str:
         return str(self.__values)
 
+    def __hash__(self) -> int:
+        if self.__hash is None:
+            h = 0
+            for key in sorted(self.__values):
+                h ^= hash(key)
+            self.__hash = h ^ len(self.__values)
```

Comment:
calculated hash is cached in `self.__hash` but `self.__values` stored a reference to the original Mapping which can be modified after hash calculated first time. 

not sure but maybe it will be better to store a copy of source mapping even if we do not take into account hash calculation

Reply:
Done. Also, I got rid of caching.